src/tools/ResultsComparer/Program.cs

```diff
@@ -185,6 +187,54 @@ private static void ExportToCsv((string id, Benchmark baseResult, Benchmark diff
             Console.WriteLine($"CSV results exported to {csvPath.FullName}");
         }
 
+        private static void ExportToXml((string id, Benchmark baseResult, Benchmark diffResult, EquivalenceTestConclusion conclusion)[] notSame, FileInfo xmlPath)
+        {
+            if (xmlPath == null)
+            {  
+                Console.WriteLine("No file given");
+                return;
+            }
+
+             if (xmlPath.Exists)
+                xmlPath.Delete();
+
+            using (XmlWriter writer = XmlWriter.Create(xmlPath.Open(FileMode.OpenOrCreate, FileAccess.Write, FileShare.Write)))
+            {
+                writer.WriteStartElement("performance-tests");
+                foreach (var slower in notSame.Where(x => x.conclusion == EquivalenceTestConclusion.Slower))
+                {
+                    writer.WriteStartElement("test");
+                    writer.WriteAttributeString("name", slower.id);
+                    writer.WriteAttributeString("type", slower.baseResult.Type);
+                    writer.WriteAttributeString("method", slower.baseResult.Method);
+                    writer.WriteAttributeString("time", "0");
+                    writer.WriteAttributeString("result", "Fail");
+                    writer.WriteStartElement("failure");
+                    writer.WriteAttributeString("exception-type", "Regression");
+                    writer.WriteElementString("message", $"{slower.id} has regressed, was {slower.baseResult.Statistics.Median} is {slower.diffResult.Statistics.Median}.");
+                    writer.WriteEndElement();
+                }
+
+                foreach (var faster in notSame.Where(x => x.conclusion == EquivalenceTestConclusion.Faster))
+                {
+                    writer.WriteStartElement("test");
+                    writer.WriteAttributeString("name", faster.id);
+                    writer.WriteAttributeString("type", faster.baseResult.Type);
+                    writer.WriteAttributeString("method", faster.baseResult.Method);
+                    writer.WriteAttributeString("time", "0");
+                    writer.WriteAttributeString("result", "Skip");
+                    writer.WriteElementString("reason", $"{faster.id} has improved, was {faster.baseResult.Statistics.Median} is {faster.diffResult.Statistics.Median}.");
+                    writer.WriteEndElement();
+                }
+
+                writer.WriteEndElement();
+                writer.Flush();
```

Comment:
good idea to flush the writer! 👍

Reply:
Doesn't flush get called as part of disposing the writer object from the using?

Reply:
TBH, I copied the flush from the code I found when I googled "C# write to xml file"

Reply:
Internet suggests that inside the using statement it isn't necessary. Anyone have any strong opinions?

Reply:
Not really.